.github/workflows/publish_tagged.yaml

```diff
@@ -17,14 +17,41 @@ env:
 
 jobs:
   tagged-deploy:
+    strategy:
+      fail-fast: true
     runs-on: ubuntu-latest
     permissions:
       contents: read
       packages: write
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-
+      # Note: The push step is at the end of this workflow, because we want to validate the binary
+      # is the right version for this tag
+      - name: Build tagged Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: false
+          tags: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ github.ref_name }}
+          build-args: |
+            gitCommit: ${{ github.sha }}
+      - name: Validate versions
+        shell: bash
+        run: |
+          set -eou pipefail
+          drand_version=$(docker run --entrypoint /bin/sh -t ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ github.ref_name }} -c "/usr/local/bin/drand -v | awk '{ printf \"v\" \$2 }'")
+          # see https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+          tag_version=$(git describe --tags | grep -P \
+          '^v(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)'\
+          '(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)'\
+          '(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?'\
+          '(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$'
+          )
+          if [[ "$drand_version" != "$tag_version" ]]; then
+            >&2 echo "Error unexpected binary version $drand_version found on for git tag $tag_version"
+            exit 1
+          fi
```

Comment:
No way.
Why is something much simpler like:
``` 
 if [ -n "$(git describe --tags | grep "$(drand -v | cut -d' ' -f2)")" ]; then
    echo "ok"
else
    echo "error"
fi
```
not okay?

Reply:
Yes that would work to ensure the binary version and one of the git tags match.

Do you want semver2 validation at any point? That ugly regex ensures that the tag is a valid semver, and thus so is the binary. The way you suggest doesn't make that assertion.